### PR TITLE
feat: add/update title of the Document

### PR DESCRIPTION
**Pull request by tanayvaswani**
Feature to add or update title to the documents. #660 

I've added the feature which provides the flexibility to users to have the custom title to their documents. 

These changes adds an input to the step form, which handles the title. See #660 for more information. 

**Added some action, types. 
Added some changes to the tRPC schema and tRPC document router.
Added some changes to the UI components.**

### CODE REVIEW

**Comment by tanayvaswani** [comment]
@Mythie Here is a draft PR with current implementation (clip on discord). Do let me know if this needs some modification. Also should we need to move this on **Add Signers'** step instead of having a dedicated step?

**Comment by Mythie** [comment]
Will check out this branch and run it shortly!

File: apps/web/src/app/(dashboard)/documents/[id]/edit-document.tsx
Comment (Mythie) [inline apps/web/src/app/(dashboard)/documents/[id]/edit-document.tsx:68]: We should also add a check here for the document status to ensure we aren't able to go back when we're out of draft mode.

File: apps/web/src/components/forms/edit-document/add-title.action.ts
Comment (Mythie) [inline apps/web/src/components/forms/edit-document/add-title.action.ts:11]: Need to check for userId here otherwise I could theoretically update anyone's document

File: packages/lib/server-only/document/update-title.ts
Comment (Mythie) [inline packages/lib/server-only/document/update-title.ts:6]: This is a mandatory option and should be reflected in the prisma query as well 😄

File: packages/trpc/server/document-router/router.ts
Comment (Mythie) [inline packages/trpc/server/document-router/router.ts:121]: Same here, you can get the user id from `ctx.user.id` 😄

File: packages/ui/primitives/document-flow/add-signers.tsx
Comment (Mythie) [inline packages/ui/primitives/document-flow/add-signers.tsx:231]: If we make the change in the place I suggested above in the review we can circumvent the logic here if we wish to.

**Comment by Mythie** [comment]
I tried it both ways in terms of design and I agree that yours feels better, I'm sure there's still work we could do to improve that experience but it's for a later time.

Once the comments above have been addressed we should be good to go 💪

File: packages/ui/primitives/document-flow/add-signers.tsx
Comment (Mythie) [reply packages/ui/primitives/document-flow/add-signers.tsx:231]: That’s fine, right now it’s not doing that though since disabled is only depending on the submission state so I guess we’d need to change that

Comment (tanayvaswani) [reply packages/ui/primitives/document-flow/add-signers.tsx:231]: I've addressed the above review (**edit-document.tsx**, first comment) of checking document status to ensure we aren't able to go back when we are out of draft mode. It does the work perfectly.

File: apps/web/src/app/(dashboard)/documents/[id]/edit-document.tsx
Comment (tanayvaswani) [reply apps/web/src/app/(dashboard)/documents/[id]/edit-document.tsx:68]: Done ✅

File: apps/web/src/components/forms/edit-document/add-title.action.ts
Comment (tanayvaswani) [reply apps/web/src/components/forms/edit-document/add-title.action.ts:11]: Yeah 👍

**Comment by Mythie** [comment]
Hey there, I think we're all good to go now! Think I just forgot to hit the approve and merge button when I was doing this the other day 😆

**Review by Mythie — APPROVED** [review]

**Comment by Mythie** [comment]
Awesome stuff!

thanks-adventure-time

**Pull request merged**
